py-polars/tests/unit/test_schema.py

```diff
@@ -225,6 +228,9 @@ def test_shrink_dtype() -> None:
         pl.Utf8,
         pl.Float32,
         pl.Boolean,
+        pl.UInt8,
```

Comment:
Polars schema must always be known statically. So we cannot change schema depending on the data. They should keep the same integer type unless we can determine statically that a smaller type would fit.

Reply:
Right, doesn't `shrink_dtype` look at the data?

Reply:
Or are you saying "since there's no data available since all are nulls, we should keep the original dtype"?

Reply:
Statically we don't know that all data is null. We only know the schema, not the data.

Reply:
Got it. I'm trying to figure out how to get the max value of the current dtype but I'm struggling a bit, can you help me out? I see `upper_bound` but I can't seem to get it to work.

Reply:
Sorry for the inconvenience. I looked better at the code and I realized we didn't adhere to the static schema logic for this function (this is one of the few exceptions).

So your logic going to smallest integer type was sound. Can you revert. Again, sorry.